### PR TITLE
Add an OCSF mapping tutorial

### PR DESCRIPTION
**Pull request by mavam**
This PR adds a tutorial on how to map events to OCSF.

### CODE REVIEW

File: web/docs/map-data-to-ocsf/README.md [hunk 1]
```diff
@@ -0,0 +1,846 @@
...
+  "tunnel_parents": null,
+  "community_id": "1:A+WZZOx8yg3UCoNV1IeiSNZUxEk=",
+  "_write_ts": null
+}
+```
+
+As we've already establisehd the mapping from conn.log to OCSF above, let's just
+write the pipeline to illustrate the different approach:
+
+```text title="conn-to-ocsf.tql"
```
Comment (dominiklohmann) [inline web/docs/map-data-to-ocsf/README.md:696]: Here's a slightly different way to structure this:

```
// tql2
read_zeek_tsv
where @name == "zeek.conn"
this = { x: this }
time = x.ts
metadata.uid = x.uid
metadata.logged_time = x._write_ts
src_endpoint.ip = x.id.orig_h
src_endpoint.port = x.id.orig_p
src_endpoint.svc_name = x.service
dst_endpoint.ip = x.id.resp_h
dst_endpoint.port = x.id.resp_p
dst_endpoint.svc_name = x.service
connection_info.uid = x.community_id
if x.local_orig and x.local_resp {
  connection_info.direction = "Lateral"
  connection_info.direction_id = 3
} else if x.local_orig {
  connection_info.direction = "Outbound"
  connection_info.direction_id = 2
} else if x.local_resp {
  connection_info.direction = "Inbound"
  connection_info.direction_id = 1
} else {
  connection_info.direction = "Unknown"
  connection_info.direction_id = 0
}
if x.id.orig_h.is_v6() or x.id.resp_h.is_v6() {
  connection_info.protocol_ver_id = 6
} else {
  connection_info.protocol_ver_id = 4
}
connection_info.protocol_name = x.proto
if x.proto == "tcp" {
  connection_info.protocol_num = 6
} else if x.proto == "udp" {
  connection_info.protocol_num = 17
} else if x.proto == "icmp" {
  connection_info.protocol_num = 1
} else {
  connection_info.protocol_num = -1
}
traffic.bytes_in = x.resp_bytes
traffic.bytes_out = x.orig_bytes
traffic.packets_in = x.resp_pkts
traffic.packets_out = x.orig_pkts
traffic.total_bytes = x.orig_bytes + x.resp_bytes
traffic.total_packets = x.orig_pkts + x.resp_pkts
// drop all the mapped fields
drop (
  x.ts, x.uid, x.id, x.service, x.local_orig, x.local_resp, x.community_id,
  x.proto, x.orig_bytes, x.resp_bytes, x.orig_pkts, x.resp_pkts, x._write_ts,
)
// move the raw event to the back and rename it to unmapped
this = { ...this, unmapped: x }
drop x
@name = "ocsf.network_activity"
```

Note that your example for the second approach (and the above rewritten one) are missing most of the OCSF event's fields.

I would argue that the best approach is in the middle between the two approaches you laid out with the following structure

```
// set input aside
this = {x: this}

// do the mapping, following the order of the OCSF schema
// …

// drop all mapped fields
drop (x.foo, x.bar, x.baz,)
this = {...this, unmapped: x}
drop x

// assign target schema name
@name = "ocsf.xyz"
```

File: web/docs/map-data-to-ocsf/README.md [hunk 2]
```diff
@@ -0,0 +1,846 @@
...
+
+The primary attributes define the semantics of the event class itself. This is
+where the core value is, as we are mapping the most event-specific information.
+
+For this, we still need to precompute several attributes that we are going to
+use in the `this = { ... }` assignment. You can see the use of `if`/`else` here
+to create a constant field based on values in the original event.
+
+```
+if local_orig == true and local_resp == true {
```
Comment (dominiklohmann) [inline web/docs/map-data-to-ocsf/README.md:372]:
```suggestion
if local_orig and local_resp {
```

This also applies in many other places below.

File: web/docs/map-data-to-ocsf/README.md [hunk 3]
```diff
@@ -0,0 +1,846 @@
...
+- **Event Class**: the description of an event defined in terms of attributes,
+  e.g., `HTTP Activity` and `Detection Finding`.
+- **Category**: a group of event classes, e.g., `System Activity` or `Findings`.
+
+The diagram below illustrates how subsets of attributes form an event class:
+
+![OCSF Event Classes](ocsf-event-classes.svg)
+
+The **Base Event Class** is a special event class that's part of every event
+class. Think of it as a mixin of attributes:
```
Comment (dominiklohmann) [inline web/docs/map-data-to-ocsf/README.md:32]: Mixin is a very technical term. Can you find something easier to understand?

File: web/docs/map-data-to-ocsf/README.md [hunk 4]
```diff
@@ -0,0 +1,846 @@
...
+benefit is that **normalization decouples data acquisition from downstream
+analytics**, allowing the processes to scale indepedently.
+
+## OCSF Primer
+
+The OCSF is a vendor-agnostic event schema (aka. "taxonomy") that defines
+structure and semantics for security events. Here are some key terms you need to
+know to map events:
+
+- **Attribute**: a unique identifier for a specific type, e.g., `parent_folder` of type `String` or `observables` of type `Observable Array`.
```
Comment (dominiklohmann) [inline web/docs/map-data-to-ocsf/README.md:22]: This could use a `gqq`

**Review by dominiklohmann — COMMENTED** [review]
This reads nicely. Some comments and an idea.

File: web/docs/map-data-to-ocsf/README.md (hunk 3 above)
Comment (mavam) [reply web/docs/map-data-to-ocsf/README.md:32]: Slightly clarified it.

**Review by dominiklohmann — APPROVED** [review]

**Pull request merged**